### DISCUSSION TITLE
Required symlinks for read-only system

### DISCUSSION THREAD

**Discussion by baileyji** [davesteele/comitup#186] [Q&A]
Hi Dave This is pretty excellent! I'm deploying it on a system that will be mostly ro, with a few things (e.g. pi-hole's db, the journal) linked from external persistent storage. For comitup to function as expected I'm sure there are some mutable config files I'll need to ensure are linked. 

I'd also like to get comitup logging with the systemd journal instead of to its own log file to facilitate log file rotation, export, and so forth.  

Could you offer some pointers in these directions?

**Comment by davesteele** [comment]
Logging is defined [here](https://github.com/davesteele/comitup/blob/edd0da01bb888324251f70acc9471308d65c97a3/comitup/comitup.py#L36) and [here](https://github.com/davesteele/comitup/blob/edd0da01bb888324251f70acc9471308d65c97a3/web/comitupweb.py#L44). Systemd offers a [JournalHandler](https://www.freedesktop.org/software/systemd/python-systemd/journal.html#journalhandler-class) to collect all log entries. This would replace the TimedRotatingFileHandler.

This has come up before. I'd consider a pull request.

Concerning RO, Comitup has an ongoing need to save DHCP leases in /var/lib/comitup/. NetworkManager requires write access to at least /etc/NetworkManager/system-connections. Also, /var/run, etc. I've not tried RO operation (SD cards are cheap) - maybe there is another PR opportunity.

You've likely seen [this](https://davesteele.github.io/raspberrypi/2021/04/29/comitup-and-pihole/), but I'll mention it anyway to avoid possible pain.
